### PR TITLE
201-vm-disk-encryption-extension patch

### PR DESCRIPTION
**Pull request by Pumpkin-3906**
- Fix 201-vm-disk-encryption-extension example

### CODE REVIEW

File: quickstart/201-vm-disk-encryption-extension/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+output "admin_ssh_key" {
+  sensitive = true
+  value     = tls_private_key.vm_key.private_key_pem
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/outputs.tf]: Could we output this private key via a [`local_file` resource](https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/file)? And we'd better include `local` provider's version restriction in `terraform` block in `providres.tf` file.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @Pumpkin-3906 for the update, I'd like to output the ssh private key via a key file so our users can ssh to the remove machine directly by using this file.

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 2]
```diff
@@ -130,3 +134,18 @@ resource "azurerm_virtual_machine_extension" "example" {
     "VolumeType"             = "All"
   })
 }
+
+resource "local_sensitive_file" "lsf" {
+  content  = tls_private_key.vm_key.public_key_openssh
+  filename = "id.rsa"
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: Please rename filename to `"id_rsa"`, thanks

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 3]
```diff
@@ -130,3 +134,18 @@ resource "azurerm_virtual_machine_extension" "example" {
     "VolumeType"             = "All"
   })
 }
+
+resource "local_sensitive_file" "lsf" {
+  content  = tls_private_key.vm_key.public_key_openssh
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: Please change to `content = tls_private_key.vm_key.private_key_pem`, thanks

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 4]
```diff
@@ -130,3 +134,18 @@ resource "azurerm_virtual_machine_extension" "example" {
...
+  filename = "id_rsa"
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: Please rename filename to `key.pem` as we'll export private key in pem format, thanks.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 5]
```diff
@@ -130,3 +134,18 @@ resource "azurerm_virtual_machine_extension" "example" {
     "VolumeType"             = "All"
   })
 }
+
+resource "local_sensitive_file" "lsf" {
+  content  = tls_private_key.vm_key.public_key_pem
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: Should be:

```hcl
content  = tls_private_key.vm_key.private_key_pem
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
@Pumpkin-3906 the content of key file is incorrect, please read the comment carefully, thanks.

File: quickstart/201-vm-disk-encryption-extension/main.tf (hunk 5 above)
Comment (Pumpkin-3906) [reply quickstart/201-vm-disk-encryption-extension/main.tf]: Fixed. Thank you.

**Review by lonegunmanb — APPROVED** [review]
Thanks @Pumpkin-3906, LGTM.

**Pull request merged**